scipy/sparse/linalg/isolve/tests/test_lsqr.py

```diff
@@ -35,6 +35,20 @@ def test_basic():
     xo = X[0]
     assert_(norm(svx - xo) < 1e-5)
 
+    # Now the same but with damp > 0.
+    # This is equivalent to solving extented system:
+    # ( G      ) * x = ( b )
+    # ( damp*I )       ( 0 )
+    damp = 1.5
+    X = lsqr(G, b, damp=damp, show=show, atol=tol, btol=tol, iter_lim=maxit)
+
+    Gext = np.r_[G, damp * np.eye(G.shape[1])]
+    bext = np.r_[b, np.zeros(G.shape[1])]
+    svx, _, _, _ = np.linalg.lstsq(Gext, bext, rcond=None)
+    xo = X[0]
+    assert_(norm(svx - xo) < 1e-5)
```

Comment:
We generally use `assert_allclose(svx, xo, atol=..., rtol=...)` for array equivalence if that is what you want to establish here.

Also since you are not using the remaining items of X (and since this codebase is pre-Py3.5, do you mind changing it to
 
```python
xo, *_ = lsqr(.....)
```

as a reminder and we will probably come back to this at some point to fix all of them later

Reply:
Touching old tests can quickly become a bottomless pit:laughing: